master/internal/resourcemanagers/resource_pool.go

```diff
@@ -375,14 +376,17 @@ func (rp *ResourcePool) receiveRequestMsg(ctx *actor.Context) error {
 		rp.getOrCreateGroup(ctx, msg.Handler).weight = msg.Weight
 
 	case job.SetGroupPriority:
-		group := rp.getOrCreateGroup(ctx, msg.Handler)
+		g := rp.getOrCreateGroup(ctx, msg.Handler)
 		if msg.Priority != nil {
-			group.priority = msg.Priority
+			g.priority = msg.Priority
+		} else {
+			ctx.Respond(fmt.Errorf("attempted to set priority of group %s to nil",
```

Comment:
these messages received here I don't think are `Ask`s are they? they' seem to still be Tells. 
specifically in case of resource pools we  might want Tell because of that forwardAll that the agent RM does.